src/pow.cpp

```diff
@@ -355,12 +355,16 @@ void multinomial_coefficients_mpz(int m, int n, map_vec_mpz &r)
 RCP<const Basic> pow_expand(const RCP<const Pow> &self)
 {
     RCP<const Basic> _base = expand(self->base_);
+    bool negative_pow = false;
     if (! is_a<Integer>(*self->exp_) || ! is_a<Add>(*_base))
         return self;
```

Comment:
Nice one. I was wondering if there is a counterexample.

Sent from my mobile phone.
On Jan 19, 2015 6:34 PM, "Isuru Fernando" <notifications@github.com> wrote:

> In src/pow.cpp
> <https://github.com/sympy/csympy/pull/373#discussion-diff-23198374>:
>
> > @@ -355,12 +355,16 @@ void multinomial_coefficients_mpz(int m, int n, map_vec_mpz &r)
> >  RCP<const Basic> pow_expand(const RCP<const Pow> &self)
> >  {
> >      RCP<const Basic> _base = expand(self->base_);
> > +    bool negative_pow = false;
> >      if (! is_a<Integer>(*self->exp_) || ! is_a<Add>(*_base))
> >          return self;
>
> Although _base is a Add, that does not guarantee that result is not a Mul
> or a Number ...
> (1/(sqrt(3)*sqrt(2) + sqrt(6)*I)**4).expand()
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sympy/csympy/pull/373/files#r23198374>.
>